Cargo.toml

```diff
@@ -23,6 +23,7 @@ glob = "0.2"
 log = "0.4.6"
 openssl = { version = '0.10.11', optional = true }
 parking_lot = "0.6"
+reqwest = "0.9.14"
```

Comment:
Could this perhaps use `curl` to use the same downloading backend as `binary-install`?

Reply:
would you mind if we did this in another PR- it's... a little tricky with the interface and i'm worried this PR is getting pretty big cc @drager

Reply:
Personally I think that would be fine, as long as we have a tracking issue for it and planning to do it in the near future.

Reply:
Ok, deferring it to an issue seems fine